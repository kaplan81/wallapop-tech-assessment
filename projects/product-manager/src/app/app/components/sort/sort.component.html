<div class="sort">
  <mat-form-field appearance="fill">
    <mat-label>Sort by</mat-label>
    <select matNativeControl required>
      <option value="volvo">Volvo</option>
      <option value="saab">Saab</option>
      <option value="mercedes">Mercedes</option>
      <option value="audi">Audi</option>
    </select>
  </mat-form-field>
</div>
