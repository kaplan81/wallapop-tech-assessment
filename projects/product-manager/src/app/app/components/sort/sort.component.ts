import { ChangeDetectionStrategy, Component } from '@angular/core';
import { MatInputModule } from '@angular/material/input';
import { MatSelectModule } from '@angular/material/select';

@Component({
  changeDetection: ChangeDetectionStrategy.OnPush,
  imports: [MatSelectModule, MatInputModule],
  selector: 'mng-sort',
  standalone: true,
  styleUrls: ['./sort.component.scss'],
  templateUrl: './sort.component.html',
})
export class SortComponent {}
