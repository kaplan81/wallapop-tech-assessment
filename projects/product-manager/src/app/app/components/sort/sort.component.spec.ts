import { ComponentFixture, TestBed } from '@angular/core/testing';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import { SortComponent } from './sort.component';

describe('SortComponent', () => {
  let fixture: ComponentFixture<SortComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [SortComponent, NoopAnimationsModule],
    }).compileComponents();
    fixture = TestBed.createComponent(SortComponent);
  });

  it('should match snapshot', () => {
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });
});
