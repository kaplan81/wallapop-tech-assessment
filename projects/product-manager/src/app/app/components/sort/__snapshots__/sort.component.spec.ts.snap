// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortComponent should match snapshot 1`] = `
<mng-sort
  __ngContext__="0"
  sortForm={[Function FormGroup]}
  sortOptions={[Function Array]}
  sortSelected={[Function EventEmitter_]}
>
  <div
    class="sort"
  >
    <form
      class="sort__form ng-untouched ng-pristine ng-valid"
      novalidate=""
    >
      <mat-form-field
        appearance="fill"
        class="mat-mdc-form-field mat-mdc-form-field-type-mat-native-select mat-form-field-no-animations mat-form-field-appearance-fill mat-primary ng-untouched ng-pristine ng-valid"
      >
        <div
          class="mat-mdc-text-field-wrapper mdc-text-field mdc-text-field--filled"
        >
          <div
            class="mat-mdc-form-field-focus-overlay"
          />
          <div
            class="mat-mdc-form-field-flex"
          >
            <div
              class="mat-mdc-form-field-infix"
            >
              <label
                class="mdc-floating-label mat-mdc-floating-label mdc-floating-label--float-above"
                matformfieldfloatinglabel=""
              >
                <mat-label>
                  Sort by
                </mat-label>
                <span
                  aria-hidden="true"
                  class="mat-mdc-form-field-required-marker mdc-floating-label--required"
                />
              </label>
              <select
                aria-invalid="false"
                aria-required="true"
                class="mat-mdc-input-element ng-untouched ng-pristine ng-valid mat-mdc-form-field-input-control mdc-text-field__input cdk-text-field-autofill-monitored"
                formcontrolname="sortCriteria"
                matnativecontrol=""
                required=""
              >
                <option
                  value="title"
                >
                   title 
                </option>
                <option
                  value="description"
                >
                   description 
                </option>
                <option
                  value="price"
                >
                   price 
                </option>
                <option
                  value="email"
                >
                   email 
                </option>
              </select>
            </div>
          </div>
          <div
            class="mdc-line-ripple mdc-line-ripple--deactivating"
            matformfieldlineripple=""
          />
        </div>
        <div
          class="mat-mdc-form-field-subscript-wrapper mat-mdc-form-field-bottom-align"
        >
          <div
            class="mat-mdc-form-field-hint-wrapper ng-trigger ng-trigger-transitionMessages"
            style="opacity:1;transform:translateY(0%);0:opacity;1:transform;"
          >
            <div
              class="mat-mdc-form-field-hint-spacer"
            />
          </div>
        </div>
      </mat-form-field>
    </form>
  </div>
</mng-sort>
`;
